web/src/lib/components/pages/profile/profile.tsx

```diff
@@ -37,99 +37,53 @@ export default class Profile extends Component<Props, State> {
   }
 
   private saveEmail() {
-    let email = this.email.value;
+    const email = this.email.value;
     this.props.user.setEmail(email);
 
     this.setState({
-      email: email,
+      email,
     });
-
-    this.render();
   }
 
   private configSendEmails(e: Event) {
-    let el = e.currentTarget as HTMLInputElement;
+    const el = e.currentTarget as HTMLInputElement;
     this.props.user.setSendEmails(el.checked);
   }
 
   private saveDemographics() {
-    let selectedIndex = this.profileAccent.selectedIndex;
-    let accent = this.profileAccent.options[selectedIndex].value;
-    this.props.user.setAccent(accent);
+    const { accent, age, gender } = this.update();
 
-    selectedIndex = this.profileAge.selectedIndex;
-    let age = this.profileAge.options[selectedIndex].value;
+    this.props.user.setAccent(accent);
     this.props.user.setAge(age);
-
-    selectedIndex = this.profileGender.selectedIndex;
-    let gender = this.profileGender.options[selectedIndex].value;
     this.props.user.setGender(gender);
 
     this.setState({
-      accent: accent,
-      age: age,
-      gender: gender,
+      accent,
+      age,
+      gender,
     });
-
-    this.render();
   }
 
   private update() {
```

Comment:
i'd love to see a return type declaration on this (as far as i can see, it would be `State`?)